src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -694,8 +701,7 @@ public Future<?> onError(Throwable e)
             state(State.FAILED); // make sure subsequent error handling sees the session in a final state 
             channel.sendControlMessage(new SessionFailedMessage()).awaitUninterruptibly();
         }
-
-        return closeSession(State.FAILED);
+        return closeSession(State.FAILED, "Failed because of an exception that is not an EOF exception;\n" + Throwables.getStackTraceAsString(e));
```

Comment:
I would just say "Failed with unknown exception;\n"

Reply:
Jon pointed out that "Failed with unknown exception" might not be accurate enough, since we know its not an EOF exception,  so I was trying to provide a better description.

Reply:
@jonmeredith can you comment here?  Personally I find this worse than "Failed with unknown exception"... sure, we know it's not EOF, but EOF is *known*....  so something *unknown* is the important part...

Reply:
I should have been clearer with my original comment, I was saying that we should include the name of the exception here - though I suppose you could work it out by the call point if you went back at the source.
```
`"Failed because of " + e.getClass().getCanonicalName() + " exception ;\n" + Throwables.getStackTraceAsString(e)`
```

Reply:
My mistake - this was unnecessary after all. I didn't realize `Throwables.getStackTraceAsString(e)` included the name which is what I cared about. Sorry to lead you astray.

Example:
```
java.lang.RuntimeException: my special ex
	at org.apache.cassandra.config.DatabaseDescriptorRefTest.testDatabaseDescriptorRef(DatabaseDescriptorRefTest.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
...
```